testing/sdk_utils.py

```diff
@@ -32,3 +33,15 @@ def gc_frameworks():
     '''Reclaims private agent disk space consumed by Mesos but not yet garbage collected'''
     for host in shakedown.get_private_agents():
         shakedown.run_command(host, "sudo rm -rf /var/lib/mesos/slave/slaves/*/frameworks/*")
+
+
+def try_throws_n_times(predicate, attempts=1, sleep_seconds=0):
+    for index in range(attempts):
+        try:
+            return predicate()
+        except:
+            if index == attempts-1:
+                raise
+            out("Encountered an exception: {}".format(traceback.format_exc()))
```

Comment:
if attempts = 3, then:

1: log exception, continue
2: log exception, continue
3: raise, don't log it